frontend/components/cards/ModuleUnionCard.tsx

```diff
@@ -22,22 +42,25 @@ const ModuleUnionCard: FC<ModuleUnionCardProps> = ({
       content = <HTMLEllipsis unsafeHTML={module.captionHtml} maxLine="3" basedOn="words" />;
       break;
     case "propositions.occurrence":
-      content = <div dangerouslySetInnerHTML={{ __html: module.title }} />;
+      content = module?.meta?.highlight?.text && ( // TODO: Replace Regex with more formal fix
```

Comment:
Looks like the `module?.meta?.highlight?.text` check appears multiple times in this file. How about putting it in a variable before the `switch` statement?
```
const snippet = module?.meta?.highlight?.text?.[0]
```
Then, the content assignments could be made more concise like this:
```
      content = snippet && (
        <HighlightEllipsis unsafeHTML={snippet} />
      );
```

Reply:
Good idea! Its been pushed!